templates/experiments/single_experiment_home.html

```diff
@@ -389,4 +504,139 @@ <h3 class="font-bold text-lg">Parent Experiments:
+          });
+        },
+        updateOperatorsByColumn(index) {
+          const filter = this.filterData.filters[index];
+          const column = filter.column;
+          if (column && this.filterData.columns[column]) {
+            filter.availableOperators = this.filterData.columns[column].operators;
+          // Set the first available operator as default
+            filter.operator = filter.availableOperators[0] || '';
+            filter.value = '';
+            filter.selectedValues = [];
+            if (this.filterData.columns[column].options) {
+              filter.filteredOptions = [...this.filterData.columns[column].options];
+            }
+          } else {
+            filter.availableOperators = [];
+            filter.operator = '';
+          }
+          this.triggerFilterChange();
+        },
+        filterDropdownOptions(index) {
+          const filter = this.filterData.filters[index];
+          const column = filter.column;
+
+          if (column && this.filterData.columns[column] && this.filterData.columns[column].options) {
+            const query = filter.searchQuery.toLowerCase().trim();
+
+            if (query === '') {
+              filter.filteredOptions = [...this.filterData.columns[column].options];
+            } else {
+              filter.filteredOptions = this.filterData.columns[column].options.filter(option =>
+                option.toLowerCase().includes(query)
+              );
+            }
+          }
+        },
+        removeFilter(index) {
+          this.filterData.filters.splice(index, 1);
+          this.triggerFilterChange();
+        },
+        updateFilterValue(index) {
+          const filter = this.filterData.filters[index];
+          if (['any of', 'all of'].includes(filter.operator)) {
+            filter.value = JSON.stringify(filter.selectedValues);
+          }
+          this.triggerFilterChange();
+        },
+        getActiveFilterCount() {
+          return this.filterData.filters.filter(filter => {
+            return filter.column && filter.operator && (
+              filter.value ||
+              (Array.isArray(filter.selectedValues) && filter.selectedValues.length > 0)
+            );
+          }).length;
+        },
+        clearAllFilters() {
+          this.filterData.filters = [];
+          this.addFilter();
+          this.filterData.showFilters = false;
+          this.triggerFilterChange();
+        },
+        triggerFilterChange() {
+          let filterParams = {};
+          this.filterData.filters.forEach((filter, i) => {
+            if (filter.column && filter.operator && (filter.value || filter.selectedValues.length > 0)) {
+              filterParams[`filter_${i}_column`] = filter.column;
+              filterParams[`filter_${i}_operator`] = filter.operator;
+              if (['any of', 'all of'].includes(filter.operator)) {
+                filterParams[`filter_${i}_value`] = JSON.stringify(filter.selectedValues);
```

Comment:
Won't `filter.value` already be the json formatted value based on the code in `updateFilterValue`